tmail-backend/tmail-third-party/openpaas/src/main/java/com/linagora/tmail/OpenPaasModule.java

```diff
@@ -0,0 +1,100 @@
+package com.linagora.tmail;
+
+import java.io.FileNotFoundException;
+
+import jakarta.inject.Named;
+import jakarta.inject.Singleton;
+
+import org.apache.commons.configuration2.Configuration;
+import org.apache.commons.configuration2.ex.ConfigurationException;
+import org.apache.james.backends.rabbitmq.RabbitMQConfiguration;
+import org.apache.james.backends.rabbitmq.RabbitMQConnectionFactory;
+import org.apache.james.backends.rabbitmq.ReactorRabbitMQChannelPool;
+import org.apache.james.backends.rabbitmq.SimpleConnectionPool;
+import org.apache.james.metrics.api.GaugeRegistry;
+import org.apache.james.metrics.api.MetricFactory;
+import org.apache.james.utils.InitializationOperation;
+import org.apache.james.utils.InitilizationOperationBuilder;
+import org.apache.james.utils.PropertiesProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+import com.google.inject.multibindings.ProvidesIntoSet;
+import com.linagora.tmail.api.OpenPaasRestClient;
+import com.linagora.tmail.configuration.OpenPaasConfiguration;
+import com.linagora.tmail.contact.OpenPaasContactsConsumer;
+
+public class OpenPaasModule extends AbstractModule {
+    private static final Logger LOGGER = LoggerFactory.getLogger(OpenPaasModule.class);
+    public static final String OPENPAAS_INJECTION_KEY = "openpaas";
+    public static final String OPENPAAS_CONFIGURATION_NAME = "openpaas";
+
+    @ProvidesIntoSet
+    public InitializationOperation initializeContactsConsumer(OpenPaasContactsConsumer instance) {
+        return InitilizationOperationBuilder
+            .forClass(OpenPaasContactsConsumer.class)
+            .init(instance::start);
+    }
+
+    @Provides
+    @Named(OPENPAAS_CONFIGURATION_NAME)
+    @Singleton
+    public Configuration providePropertiesConfiguration(PropertiesProvider propertiesProvider) throws ConfigurationException {
+        try {
+            return propertiesProvider.getConfiguration(OPENPAAS_CONFIGURATION_NAME);
+        } catch (FileNotFoundException e) {
+            LOGGER.error("Could not find configuration file '{}.properties'",
+                OPENPAAS_CONFIGURATION_NAME);
+            throw new RuntimeException(e);
```

Comment:
Please just return disabled here: this is optional.

Reply:
What is optional? we are loading the OpenPaasModule using a module chooser, this code is only executed if we explicitly enable the module, now that the fallback is not implemented, we fail when the openpaas.properties file is not found.